### PR TITLE
Raise appropriate errors in keras.layers.Layer.output and input.

### PR DESCRIPTION
**Pull request by dibgerge**
**Describe the current behavior**

When a layer has multiple nodes (shared layer) in `tensorflow.python.keras`, calling the method `Layer.output` or `Layer.output` returns only the first node created.

**Describe the expected behavior**
As per the `keras` API and current documentation, the two properties should fail with an `AttributeError`. In addition, those two properties should fail if tensorflow is executing eagerly.

### CODE REVIEW

**Comment by dibgerge** [comment]
Signed CLA.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @fchollet: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Review by fchollet — APPROVED** [review]
Thanks for the PR. LGTM

**Comment by fchollet** [comment]
Due to recent changes, this PR is no longer valid:

- We want to always return the first output as `layer.output`.
- `layer.output` is meaningful in eager.
